core/src/main/java/org/commonjava/indy/core/content/DefaultContentManager.java

```diff
@@ -373,7 +391,7 @@ public Transfer store( final ArtifactStore store, final String path, final Input
 
             contentGeneratorManager.handleContentStorage( transferStore, path, txfr, eventMetadata );
 
-            clearNFCEntries(kl, path);
+            nfcContentCleanExecutor.execute( () -> clearNFCEntries( kl, path ) );
```

Comment:
I wonder if we should move the clearNFCEntries method to the NotFoundCache implementation? It seems like it would encapsulate the NFC behavior better that way...WDYT?

Reply:
Agree. So it means some modification in galley is needed.

Reply:
I guess we can make that change in galley after the currently pending release, since it's not really urgent.